src/components/map/components/DeparturesDialogSheet.tsx

```diff
@@ -43,17 +43,82 @@ const DeparturesDialogSheet = ({
     option: 'now',
     date: new Date().toISOString(),
   });
-  const [showOnlyFavorites, setShowOnlyFavorites] = useState<boolean>(false);
-
-  const message = t(NearbyTexts.results.messages.emptyResult);
   const [longitude, latitude] = stopPlaceFeature.geometry.coordinates;
-  const {closestLocation, isSearching} = useReverseGeocoder(
-    {longitude, latitude} || null,
+  const {
+    closestLocation,
+    isSearching: isGeocoderSearching,
+    error: geocoderError,
+  } = useReverseGeocoder({longitude, latitude} || null);
+
+  const {state: stopDetailsState} = useStopsDetailsData(
+    closestLocation && [closestLocation.id],
   );
+  const {
+    data: stopDetailsData,
+    isLoading: isStopDetailsLoading,
+    error: stopDetailsError,
+  } = stopDetailsState;
+
+  const stopPlace = stopDetailsData?.stopPlaces?.[0];
+  const isLoading = isStopDetailsLoading || isGeocoderSearching;
+  const didLoadingDataFail = !!geocoderError || !!stopDetailsError;
+
+  const memoizedComponents = useMemo(() => {
```

Comment:
Why use `useMemo` here? Seems a little strange as there are no expensive calculations. From the React doc for ùseMemo`:
> This optimization helps to avoid expensive calculations on every render.

Reply:
Is not expensive but avoids re-calculation 😅, I took the concept from other places like `src/components/bottom-sheet/BottomSheetContext.tsx`, do you want me to change it to a function instead?

Reply:
In the bottom-sheet there is some technicality with calculating the animation and stuff like that which makes it necessary. I think here we can just skip `useMemo`. Using `useMemo` is not without risk, as dependencies can be forgotten causing bugs in the implementation.

Reply:
Fixed :)